main.go

```diff
@@ -79,44 +97,46 @@ func logic(ctx context.Context, o opts) error {
 		return fmt.Errorf("error joining gus server url: %w", err)
 	}
 
+	plugins := make(map[string]plugin)
+	if err := loadPlugin(plugins, o.plugin, o.pluginArgs); err != nil {
+		return fmt.Errorf("error loading plugin %s: %w", o.plugin, err)
+	}
+
 	gusCfg := gusapi.NewConfiguration()
 	gusCfg.BasePath = gusBasePath
 	gusCli := gusapi.NewAPIClient(gusCfg)
 
 	if o.skipWaiting {
-		log.Print("skipping waiting, performing an immediate updateProcess")
-		if err := updateProcess(ctx, gusCli, machineID, o.gusServer, sbomHash, o.destinationDir, httpPassword, httpPort); err != nil {
+		log.Print("skipping waiting, performing an immediate update check")
+		if err := updateProcess(ctx, gusCli, plugins, machineID, o.gusServer, sbomHash, o.destinationDir, httpPassword, httpPort); err != nil {
 			// If the updateProcess fails we exit with an error
 			// so that gokrazy supervisor will restart the process.
 			return fmt.Errorf("error performing updateProcess: %v", err)
 		}
-
-		// If the updateProcess doesn't error
-		// we happily return to terminate the process.
-		return nil
 	}
 
-	log.Print("entering update checking loop")
-	ticker := time.NewTicker(frequency)
-
-	for {
+	for c := time.Tick(frequency); ; {
```

Comment:
Changing this so then we get an immediate first check without waiting an extra `frequency` interval.

Also changing the skipWaiting to only skip the Jitter once, the first time.

These two changes combined still achieve the same initially defined behaviour for skipWaiting, but improve the first check also in cases where skipWaiting is set to `false`